modules/integration/tests-integration/tests-metadata/tests-metadata-generic/src/test/java/org/wso2/carbon/registry/metadata/test/restservice/RESTServiceCreationTestCase.java

```diff
@@ -0,0 +1,336 @@
+        GenericArtifact artifact = artifactManager.newGovernanceArtifact(new QName("RESTService5"));
+
+        artifact.setAttribute("overview_name", "RESTService5");
+        artifact.setAttribute("overview_context", "/rs_test");
+        artifact.setAttribute("overview_version", "1.0.1");
+        artifact.setAttribute("interface_swagger", "http://petstore.swagger.io/v2/swagger.json");
+        artifact.setAttribute("endpoints_entry", "None:http://test.org");
+
+        artifactManager.addGenericArtifact(artifact);
+
+        GenericArtifact receivedArtifact = artifactManager.getGenericArtifact(artifact.getId());
+        assertNotNull(receivedArtifact.getDependencies());
+        assertEquals(receivedArtifact.getDependencies().length, 3, "Expecting 2 endpoint dependencies and swagger.");
+
+        artifactManager.removeGenericArtifact(artifact.getId());
+    }
+
+    /**
+     * Create rest service artifact including wadl url and an endpoint. expected to be having 3 associations
+     * for the created service which are wadl resource and the endpoint created and also the endpoint added manually.
+     *
+     * @throws GovernanceException If fails to create the rest service.
+     */
+
+    @Test(groups = { "wso2.greg" }, description = "add REST service with endpoints and wadl")
+    public void createRESTServiceWithEndpointsAndWadl() throws GovernanceException {
+        GenericArtifact artifact = artifactManager.newGovernanceArtifact(new QName("RESTService6"));
+
+        artifact.setAttribute("overview_name", "RESTService6");
+        artifact.setAttribute("overview_context", "/rs_test");
+        artifact.setAttribute("overview_version", "1.4.1");
+        artifact.setAttribute("interface_wadl",
+                "https://raw.githubusercontent.com/wso2/wso2-qa-artifacts/master/automation-artifacts/greg/wadl/SearchSearvice.wadl");
+        artifact.setAttribute("endpoints_entry", "None:http://test.org");
+
+        artifactManager.addGenericArtifact(artifact);
+
+        GenericArtifact receivedArtifact = artifactManager.getGenericArtifact(artifact.getId());
+        assertNotNull(receivedArtifact.getDependencies());
+        assertEquals(receivedArtifact.getDependencies().length, 3, "Expecting 3 endpoint dependencies and swagger.");
+
+        artifactManager.removeGenericArtifact(artifact.getId());
+    }
+
+    /**
+     * Create rest service artifact including an invalid swagger url. Expected to throw an exception.
+     *
+     * @throws GovernanceException If fails to create the rest service.
+     */
+    @Test(groups = { "wso2.greg" }, description = "add REST service with invalid swagger url.",
+            expectedExceptions = GovernanceException.class)
+    public void createRESTServiceWithInvalidSwaggerUrl() throws GovernanceException {
+        GenericArtifact artifact = artifactManager.newGovernanceArtifact(new QName("RESTService7"));
+
+        artifact.setAttribute("overview_name", "RESTService7");
```

Comment:
Isn't it better to use constants for keys such as "overview_name" since those have appeared multiple times?